dbus_dump_watcher.cpp

```diff
@@ -0,0 +1,141 @@
+{
+    sdbusplus::message::object_path objPath;
+    DBusInteracesMap interfaces;
+    msg.read(objPath, interfaces);
+    std::string strObjPath = objPath;
+    log<level::ERR>(
+        fmt::format("DBusDumpWatcher::interfaceAdded path ({})", strObjPath)
+            .c_str());
+    auto iter = interfaces.find(_entryIntf);
+    if (iter == interfaces.end())
+    {
+        // ignore not specific to the dump type being watched
+        return;
+    }
+    uint32_t id = std::stoul(objPath.filename());
+    _entryPropWatchList.emplace(
+        strObjPath, std::make_unique<sdbusplus::bus::match_t>(
+                        _bus,
+                        sdbusplus::bus::match::rules::propertiesChanged(
+                            strObjPath, progressIntf),
+                        [this, strObjPath, id](auto& msg) {
+                            this->propertiesChanged(strObjPath, id, msg);
+                        }));
+}
+
+void DBusDumpWatcher::interfaceRemoved(sdbusplus::message::message& msg)
+{
+    sdbusplus::message::object_path objPath;
+    DBusInteracesMap interfaces;
+    msg.read(objPath, interfaces);
+    std::string strObjPath = objPath;
+    log<level::ERR>(
+        fmt::format("DBusDumpWatcher::interfaceRemoved path ({})", strObjPath)
+            .c_str());
+    auto iter = interfaces.find(_entryIntf);
+    if (iter == interfaces.end())
+    {
+        // ignore not specific to the dump type being watched
+        return;
+    }
+    _entryPropWatchList.erase(strObjPath);
+}
+
+void DBusDumpWatcher::propertiesChanged(const std::string& objPath, uint32_t id,
+                                        sdbusplus::message::message& msg)
+{
+    std::string interface;
+    DBusPropertiesMap propMap;
+    msg.read(interface, propMap);
+    log<level::INFO>(
+        fmt::format("propertiesChanged objectpath ({}) id ({})", objPath, id)
+            .c_str());
+
+    bool fcomplete = isDumpProgressCompleted(propMap);
+    if (!fcomplete)
+    {
+        log<level::INFO>(
+            fmt::format("propertiesChanged objectpath ({}) status is not "
+                        "complete",
+                        objPath)
+                .c_str());
+        return;
+    }
+
+    uint64_t size = getDumpSize(_bus, objPath);
+    openpower::dump::pldm::sendNewDumpCmd(id, _dumpType, size);
+
+    _entryPropWatchList.erase(objPath);
+}
+
+void DBusDumpWatcher::addInProgressDumpsToWatch()
+{
+    ManagedObjectType objects = openpower::dump::getDumpEntries(_bus);
+    for (auto& object : objects)
+    {
+        auto iter = object.second.find(_entryIntf);
+        if (iter == object.second.end())
+        {
+            continue;
+        }
+        bool fcomplete = isDumpProgressCompleted(object.second);
+        if (!fcomplete)
+        {
+            std::string objectPath = object.first;
+            uint32_t id = std::stoul(object.first.filename());
+            log<level::INFO>(
```

Comment:
Do we need this trace?

Reply:
yes want to check, there could be some dumps which were initiated but the debug-collector crashed. probably they need to do some cleanup